azure-pipelines.yml

```diff
@@ -284,6 +308,7 @@ jobs:
     - sw_build
     - top_earlgrey_verilator
     - top_earlgrey_nexysvideo
+    - riscv_compliance_tests
```

Comment:
That dependency is unnecessary.

Reply:
my original rationale was the same as above..i was thinking that we wouldn't want to do a release unless the testing phases all worked (i was actually wondering why the normal verilator tests werent there either).

I'll take it for now, but do let me know what you think.

Reply:
Dependencies are only needed for correctness. The overall pass/fail status of the pipeline isn't influenced by that.

In this case, we want to upload artifacts as soon as we can. You want to be able to use the CI-generated artifacts and do local debugging on them, e.g. to figure out why a test in CI failed.

On the point of releasing only binaries which have passed CI: we currently do that by first running CI on a specific commit on master, and then tagging that commit. That tagging will re-run CI and, assuming the output is deterministic, we get a working release. That's not ideal, but good enough and actually pretty common. If you want to get things fully correct, you'd want to "promote" release candidate builds to releases after they have passed testing. That requires a couple tricks to get the version in the binaries and the git tagging correct. Let's save that for another day.

Reply:
o actually, i did not know we could download the CI generated artifacts for local debugging. I thought we could only download the released versions, am I missing something?

In any case, for this case, I've removed the dependency.

Reply:
If you click on any Azure build, e.g. https://dev.azure.com/lowrisc/opentitan/_build/results?buildId=6361&view=results, click on the top-right Artifacts button, and download all outputs that we produce. The opentitan-dist one is the file that will be uploaded to GH releases if we do a tagged release.

Reply:
thanks @imphil !